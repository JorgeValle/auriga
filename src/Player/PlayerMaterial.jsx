import React from 'react';

import King from '../Pieces/King';
import Queen from '../Pieces/Queen';
import Bishop from '../Pieces/Bishop';
import Rook from '../Pieces/Rook';
import Pawn from '../Pieces/Pawn';

// Components should represent one piece of data model
class PlayerMaterial extends React.Component {
  render() {

    const player = this.props.player;

    return (
      <div className="player-material">
        <span>Pieces</span>
        <meter min="0"  value="13" max="16"></meter>
        <span>13 / 16</span>
  
        <div>
          <King />
          <Queen />
          <Bishop />
          <Rook />
          <Pawn />
        </div>

      </div>
    );
  }
}

export default PlayerMaterial;