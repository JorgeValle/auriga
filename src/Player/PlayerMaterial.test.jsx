import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import { shallow, configure } from 'enzyme';
import PlayerMaterial from './PlayerMaterial.jsx';

configure({
  adapter: new Adapter()
});

describe('<PlayerMaterial />', () => {

  // We shallow render to make sure changes in children don't cause cascading fails
  it('shallow renders', () => {
    const wrapper = shallow(<PlayerMaterial />);
  });

  // Snapshot matching
  it('matches snapshot', () => {
    const wrapper = shallow(<PlayerMaterial />);
    expect(wrapper).toMatchSnapshot();
  });

});