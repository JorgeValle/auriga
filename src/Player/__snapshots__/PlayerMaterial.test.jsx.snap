// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PlayerMaterial /> matches snapshot 1`] = `ShallowWrapper {}`;
